### PR TITLE
feat(client): Implement new api & optimize code structure

### PR DESCRIPTION
**Pull request by afterimagex**
- Enhanced Naming Convention Support: Introduced support for underscore-named function calls in addition to existing camelCase naming convention. This allows for greater flexibility and compatibility with different coding styles and external libraries.

- Expanded index_label Types: Augmented the index_label functionality by adding support for shard and unique types. These additional types provide enhanced indexing capabilities that enable more sophisticated data organization and retrieval scenarios, improving overall performance and scalability.

- New Rank API Implementation: Implemented a new rank API to support ranking and sorting of data based on specified criteria. This feature enables applications to easily incorporate ranking logic into their workflows, enhancing user experience by presenting the most relevant or prioritized information first.

- Optimized HTTP Invocation and Response Handling: Refactored HTTP client invocations to utilize a more streamlined approach. Introduced ResponseValidation for managing response checks, effectively reducing the need for explicit check_if_success calls in multiple locations. This optimization not only simplifies code but also enhances its maintainability and readability, making it easier to identify and resolve potential issues related to HTTP response handling.

TODO:
- handle some py-lint problems before Merge to Master/Main

### CODE REVIEW

File: hugegraph-python-client/src/pyhugegraph/utils/log.py [hunk 1]
```diff
@@ -46,7 +48,7 @@ class CustomConsoleHandler(logging.StreamHandler):
             "INFO": "\033[0;32m",  # Green
             "WARNING": "\033[0;33m",  # Yellow
             "ERROR": "\033[0;31m",  # Red
-            "CRITICAL": "\033[0;41m"  # Red background
+            "CRITICAL": "\033[0;41m",  # Red background
```
Comment (imbajin) [inline hugegraph-python-client/src/pyhugegraph/utils/log.py:51]: Extra  `,`?

File: hugegraph-python-client/README.md
Comment (imbajin) [inline hugegraph-python-client/README.md:1]: Could also add the cmd for
```markdown
### Install from source
release soon
```
image

File: hugegraph-python-client/README.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ release soon
 ```python
 from pyhugegraph.client import PyHugeClient
 
-client = PyHugeClient("127.0.0.1", "8080", user="admin", pwd="admin", graph="hugegraph")
+client = PyHugeClient("127.0.0.1", "8080", user="admin", pwd="admin", graph="hugegraph", gs="DEFAULT")
```
Comment (imbajin) [inline hugegraph-python-client/README.md:20]: - add a comment for this line (like `gs` is optional)
- currently `gs` compatible with `graphspace`? (set as alias)

File: hugegraph-python-client/src/pyhugegraph/utils/log.py (hunk 1 above)
Comment (afterimagex) [reply hugegraph-python-client/src/pyhugegraph/utils/log.py:51]: black-formatter

Comment (imbajin) [reply hugegraph-python-client/src/pyhugegraph/utils/log.py:51]: > black-formatter

OK, just keep it

**Pull request merged**